### PR TITLE
Adding github authentication guide to lab 1

### PR DESCRIPTION
**Pull request by Heaper327**
Adding instructions to Lab 1 on how to authenticate git without plaintext password, since many students encountered problems with cloning / pulling / pushing to their repo.

Specifically, adding instructions on how to create and authorize PA tokens, how to use `gh`, and link to github doc on authentication options.

### CODE REVIEW

File: _docs/lockpicking_lab.md [hunk 1]
```diff
@@ -49,7 +49,44 @@ Then checkout your repository as follows:
...
+remote: Support for password authentication was removed on August 13, 2021.
+remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
+fatal: Authentication failed for 'https://github.com/illinois-cs-coursework/{{site.data.constants.semester }}_{{ site.data.constants.department_code }}{{site.data.constants.course_number}}_NETID.git'
+```
+
+As stated by the  message implies, Github no longer allows you to login with plaintext password due to security concerns. There are several ways around this, we recommend using Github's command line tool `gh`, or using a personal access token:
+### gh
+
+`gh` is the offical command-line interface for Github. You will first need to install it using `apt`. Be sure to install it as root:
+```console
```
Comment (nd-0r) [inline _docs/lockpicking_lab.md:64]: Probably should `sudo apt update` first

**Review by nd-0r — CHANGES_REQUESTED** [review]
Looks good to me. I think this will help a lot.

**Review by xwei16 — COMMENTED** [review]
One more thing we may want to mention: if they are on VSCode, sometimes they have to sign out then sign in after generating the token.

**Review by nd-0r — APPROVED** [review]

**Review by xwei16 — APPROVED** [review]
Lgtm!

**Pull request merged**